### PR TITLE
Contrast parser: switch endpoint to unsaved_endpoints

### PR DESCRIPTION
**Pull request by damiencarol**
## Work
- [x] switch endpoint to unsaved_endpoints
- [x] enhance de-duplication
- [x] add more unit tests

### CODE REVIEW

File: dojo/tools/contrast/parser.py [hunk 1]
```diff
@@ -36,45 +37,47 @@ def get_findings(self, filename, test):
...
+                description=description,
+                severity=severity,
+                numerical_severity=Finding.get_numerical_severity(severity),
+                dynamic_finding=True,
+                static_finding=False,
+                vuln_id_from_tool=row.get('Rule Name'),
+                unique_id_from_tool=row.get('Vulnerability ID'),
+                nb_occurences=1,
+            )
+            endpoint = Endpoint(host="0.0.0.0", protocol="http")
```
Comment (valentijnscholten) [inline dojo/tools/contrast/parser.py:55]: is this something we want to keep, http://0.0.0.0 as default?

Comment (aaronweaver) [reply dojo/tools/contrast/parser.py:55]: Probably not.

Comment (damiencarol) [reply dojo/tools/contrast/parser.py:55]: it was the value in the current code. so, ... what do we do??? :D

Comment (valentijnscholten) [reply dojo/tools/contrast/parser.py:55]: bring it inline with what we usually do, but there is this other big PR around endpoints coming anyway.

Comment (damiencarol) [reply dojo/tools/contrast/parser.py:55]: @aaronweaver @valentijnscholten I don't have a better option than keep the same behavior. Also this PR  fix de-duplication not only the endpoints. I already discussed with the author of the other big PR, it's not a pb to resolve conflits on his side.

Comment (valentijnscholten) [reply dojo/tools/contrast/parser.py:55]: maybe just don't add an endpoint if 'Request URI' is not there?

Comment (damiencarol) [reply dojo/tools/contrast/parser.py]: made this modification and updated unit tests

Comment (valentijnscholten) [reply dojo/tools/contrast/parser.py]: Thanks, just looked again and I thought the URI contained the hosts but there is no host so they use 0.0.0.0.
This is an interesting case as ideally the uploader should supply an endpoint in the upload and that should be used to get the hostname? Not for this PR maybe, but something to think about when improving/changing the parser contract.

**Review by mtesauro — APPROVED** [review]
Approved

**Review by valentijnscholten — APPROVED** [review]

**Pull request merged**